packages/vm/test/t8n/transitionTool.ts

```diff
@@ -0,0 +1,381 @@
+import { Block } from '@ethereumjs/block'
+import { EVMMockBlockchain, MCLBLS } from '@ethereumjs/evm'
+import { RLP } from '@ethereumjs/rlp'
+import { TransactionType, TxData, createTxFromTxData } from '@ethereumjs/tx'
+import {
+  Account,
+  BIGINT_1,
+  CLRequestType,
+  bigIntToHex,
+  createAddressFromString,
+  hexToBytes,
+  toBytes,
+  zeros,
+} from '@ethereumjs/util'
+import { keccak256 } from 'ethereum-cryptography/keccak'
+import { bytesToHex } from 'ethereum-cryptography/utils'
+import { readFileSync } from 'fs'
+import { loadKZG } from 'kzg-wasm'
+import * as mcl from 'mcl-wasm'
+
+import { buildBlock } from '../../dist/esm/buildBlock.js'
+import { createVM } from '../../dist/esm/constructors.js'
+import { getCommon } from '../tester/config.js'
+import { makeBlockFromEnv, setupPreConditions } from '../util.js'
+
+import { normalizeNumbers } from './helpers.js'
+
+import type { PostByzantiumTxReceipt } from '../../dist/esm/types.js'
+import type { VM } from '../../dist/esm/vm.js'
+import type { RunnerOptions, T8NAlloc, T8NEnv, T8NOptions } from './types.js'
+import type { Common } from '@ethereumjs/common'
+import type { Log } from '@ethereumjs/evm'
+import type { TypedTxData } from '@ethereumjs/tx'
+import type { Address } from '@ethereumjs/util'
+
+export class TransitionTool {
+  public options: T8NOptions
+
+  public alloc: T8NAlloc
+  public txsData: TypedTxData[]
+  public inputEnv: T8NEnv
+
+  public common!: Common
+  public stateTracker!: StateTracker
+
+  private constructor(args: T8NOptions) {
+    this.options = args
+
+    this.alloc = JSON.parse(readFileSync(args.input.alloc).toString())
+    this.txsData = JSON.parse(readFileSync(args.input.txs).toString())
+    this.inputEnv = normalizeNumbers(JSON.parse(readFileSync(args.input.env).toString()))
+  }
+
+  static async init(args: T8NOptions, runnerOptions: RunnerOptions = {}) {
+    const t8nTool = new TransitionTool(args)
+    t8nTool.common = getCommon(args.state.fork, await loadKZG())
+
+    const common = t8nTool.common
+
+    const blockchain = new EVMMockBlockchain()
+
+    // Override `getBlock` to ensure `BLOCKHASH` works as expected (reads from `inputEnv.blockHashes`)
+    blockchain.getBlock = async function (number?: Number) {
+      for (const key in t8nTool.inputEnv.blockHashes) {
+        if (Number(key) === number) {
+          return {
+            hash() {
+              return hexToBytes(t8nTool.inputEnv.blockHashes[key])
+            },
+          }
+        }
+      }
+      // Hash not found, so return the zero hash
+      return {
+        hash() {
+          return zeros(32)
+        },
+      }
+    }
+
+    await mcl.init(mcl.BLS12_381)
+    const bls = new MCLBLS(mcl)
```

Comment:
Don't know, do we want to use MCL here? Or do we rather also want to use the pure/default JS version?

Reply:
I wonder though if we should rather generally move over to WASM with this (so basically the setup we have in the client), since the tool atm still is relatively slow and this is already noticeable, e.g. on filling the 1153 tests, this already takes a solid amount of time.

Maybe we also want to decide where it's worth to switch - e.g. for very basic and settled stuff like `keccak` or `ecrecover` we can very likely do (?) - and for other things (e.g. Noble/MCL for BLS might be a case here) we might rather want to test/fill with our native code.

Thoughts? Opinions?